runtime/src/topo/hwloc/topo-hwloc.c

```diff
@@ -172,9 +173,29 @@ void chpl_topo_init(void) {
   //
   // Find the NUMA nodes, that is, the objects at numaLevel that also
   // have CPUs.  This is as opposed to things like Xeon Phi HBM, which
-  // is memory-only, no CPUs.
+  // is memory-only, no CPUs.  Allow for overriding this through the
+  // environment.
   //
-  {
+  if (strcmp(CHPL_LOCALE_MODEL, "flat") != 0) {
+    //
+    // The number of NUMA domains only matters for locale models other
+    // than 'flat'.
+    //
+    numNumaDomains = (int) chpl_env_rt_get_uint("NUM_NUMA_DOMAINS", 0);
+    if (numNumaDomains != 0) {
+      //
+      // If you're using Qthreads tasking, forcing the number of NUMA
+      // domains also forces the number of Qthreads shepherds.  (If we
+      // don't do this we can find ourselves trying to queue tasks on
+      // nonexistent shepherds.)
+      //
+      if (strcmp(CHPL_TASKS, "qthreads") == 0) {
```

Comment:
This bit seems a little odd to me. I wonder if it would be cleaner to have this code in qthreads doing something like:

```c
if (chpl_topo_getNumNumaDomains() > hwpar) { 
  hwpar = chpl_topo_getNumNumaDomains();
 }
```

Reply:
Yeah, I wondered about that too.  The only downside is that it spreads the effect of `CHPL_NUM_NUMA_DOMAINS` across multiple source files without an obvious connection.  On the other hand I guess it fixes a potential problem, in that even now with none of this work in yet, someone running with the numa locale model and a very restricted thread count could get internal errors as a result!  I'll do as you describe.